presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/RewriteIfOverAggregation.java

```diff
@@ -0,0 +1,244 @@
+        }
+        return plan;
+    }
+
+    // Map<VariableReferenceExpression, RowExpression> stores the candidate IF expressions for rewrite.
+    private static class Rewriter
+            extends SimplePlanRewriter<Map<VariableReferenceExpression, RowExpression>>
+    {
+        private final PlanVariableAllocator planVariableAllocator;
+        private final PlanNodeIdAllocator planNodeIdAllocator;
+
+        private Rewriter(PlanVariableAllocator variableAllocator, PlanNodeIdAllocator idAllocator)
+        {
+            planVariableAllocator = variableAllocator;
+            planNodeIdAllocator = idAllocator;
+        }
+
+        private static boolean isCandidateIfExpression(RowExpression rowExpression)
+        {
+            return rowExpression instanceof SpecialFormExpression && ((SpecialFormExpression) rowExpression).getForm().equals(IF)
+                    && ((SpecialFormExpression) rowExpression).getArguments().get(2) instanceof ConstantExpression
+                    && ((ConstantExpression) ((SpecialFormExpression) rowExpression).getArguments().get(2)).isNull()
+                    && ((SpecialFormExpression) rowExpression).getArguments().get(1) instanceof VariableReferenceExpression;
+        }
+
+        private static VariableReferenceExpression getTrueValueFromIf(RowExpression rowExpression)
+        {
+            checkState(rowExpression instanceof SpecialFormExpression
+                    && ((SpecialFormExpression) rowExpression).getArguments().get(1) instanceof VariableReferenceExpression);
+            return (VariableReferenceExpression) ((SpecialFormExpression) rowExpression).getArguments().get(1);
+        }
+
+        @Override
+        public PlanNode visitPlan(PlanNode node, RewriteContext<Map<VariableReferenceExpression, RowExpression>> context)
+        {
+            // This optimizer targets plan generated from IF(predicate, AGG(x)). The plan generated will be Project <- Aggregation, where project includes the IF expression.
+            // Here we pass an empty map in context by default, so as to not pass the candidate IF expressions unless it's a project node.
+            return context.defaultRewrite(node, ImmutableMap.of());
+        }
+
+        @Override
+        public PlanNode visitProject(ProjectNode node, RewriteContext<Map<VariableReferenceExpression, RowExpression>> context)
+        {
+            // The rewrite will change the aggregation output, hence we only do the rewrite if the aggregation output is only used in the IF expression, i.e. variables which
+            // occurs only once in the assignments.
+            Set<VariableReferenceExpression> candidateVariables = node.getAssignments().getExpressions().stream()
+                    .flatMap(expression -> extractAll(expression).stream())
+                    .collect(Collectors.groupingBy(identity(), Collectors.counting()))
+                    .entrySet()
+                    .stream()
```

Comment:
nit: we break line only after `.stream()`
```java
.collect(Collectors.groupingBy(identity(), Collectors.counting())).entrySet().stream()
...

Reply:
Done